app/models/piece.rb

```diff
@@ -19,6 +19,20 @@ def move_to!(new_row, new_col)
     update(row_position: new_row, col_position: new_col, moved: true)
   end
 
+  def moving_into_check?(row_dest, col_dest)
+    row_pos = row_position
```

Comment:
I think @ronny2205 put them there to save the initial variable - once we update the coordinates we can't access the initial coordinates (unless we use your suggested assign_attributes :smile: )